### PR TITLE
React/week4 HW

### PR DESCRIPTION
**Pull request by rajanibusani**

### CODE REVIEW

File: react/week4/github-user-app/src/components/UserContext.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+export const UserContext = createContext();
+
+export const UserProvider = ({ children }) => {
+    const [isLoading, setIsLoading] = useState(false);
+    const [users, setUsers] = useState([]);
+    const [userSearch, setUserSearch] = useState("");
+    const [error, setError] = useState("");
+    const [repos, setRepos] = useState([]);
+
+    const fetchUsers = async (url) => {
```
Comment (obelitz) [inline react/week4/github-user-app/src/components/UserContext.js:11]: I'd put the `API_URL` somewhere in this file, maybe even this method. 
Right now this method fetches anything you pass in via the `url` param.

File: react/week4/github-user-app/src/components/UserContext.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        try {
+            const fetchUsers = await fetch(url)
+            const usersData = await fetchUsers.json();
+            console.log(usersData)
```
Comment (obelitz) [inline react/week4/github-user-app/src/components/UserContext.js:15]: I would usually remove `console.log`s before pushing my code, so the end-user does not see data in their console

File: react/week4/github-user-app/src/components/UserContext.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+            if (usersData.items) {
+                setIsLoading(false)
+                return setUsers(usersData.items)
+            } else if (usersData.message) {
+                console.log(error)
+                setIsLoading(false)
+                return setError([usersData.message]);
```
Comment (obelitz) [inline react/week4/github-user-app/src/components/UserContext.js:22]: Why are you setting the error as an array? :)

File: react/week4/github-user-app/src/components/UserContext.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+            }
+        } catch (error) {
+            setError(error);
```
Comment (obelitz) [inline react/week4/github-user-app/src/components/UserContext.js:25]: Remember to reset `isLoading` down here as well

File: react/week4/github-user-app/src/components/UserContext.js (hunk 3 above)
Comment (obelitz) [reply react/week4/github-user-app/src/components/UserContext.js:22]: If you know this is an error, you can always `throw new Error(usersData.message)` and then let the `catch` block handle it

File: react/week4/github-user-app/src/components/UserContext.js [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    const UserValues = {
+        users,
+        userSearch,
+        isLoading,
+        error,
+        getUsers: (url) => {
+            if (userSearch) {
+                console.log(userSearch)
+                setIsLoading(true);
+                return fetchUsers(url);
```
Comment (obelitz) [inline react/week4/github-user-app/src/components/UserContext.js:39]: I'd move this logic into the `fetchUsers` method

File: react/week4/github-user-app/src/components/UserContext.js [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+            }
+        },
+        onChange: (e) => {
+            setUserSearch(e.target.value)
+        },
+        getRepos: async (url) => {
+            const fetchRepos = await fetch(url);
+            const reposData = await fetchRepos.json();
+            return setRepos(reposData);
```
Comment (obelitz) [inline react/week4/github-user-app/src/components/UserContext.js:48]: I'd also have the HYF repo-url in this method :)

File: react/week4/github-user-app/src/components/UserHandler.js [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+    }, [userContext.userSearch])
+
+    return (
+        <div>
+            {userContext.isLoading && <p>Searching...</p>}
+            { userContext.error !== "" && <h3>{userContext.error}</h3>}
+            { (userContext.users.length > 0) ? <UserList users={userContext.users} />
+                : userContext.userSearch && <h3>NO USERS FOUND</h3>}
+        </div>
+    )
```
Comment (obelitz) [inline react/week4/github-user-app/src/components/UserHandler.js:23]: Remember you can have multiple `return` statements in a component:
```
if(userContent.isLoading) return <p>Searching</p>
if(userContent.error !== "") return <h3>{userContext.error}</h3>
if(userContent.users.length > 0 ) return <...
```

File: react/week4/github-user-app/src/components/UserInput.js [hunk 8]
```diff
@@ -0,0 +1,10 @@
+import React, { useContext } from "react";
+import { UserContext } from "./UserContext";
+const UserInput = () => {
+    const userContext = useContext(UserContext);
+    return <>
+        <input type="text" placeholder="search username" value={userContext.userSearch} onChange={userContext.onChange}></input>
+    </>
```
Comment (obelitz) [inline react/week4/github-user-app/src/components/UserInput.js:7]: You don't need the fragment `<>` here.

**Review by obelitz — COMMENTED** [review]
Good job :)

**Comment by rajanibusani** [comment]
> Good job :)

Thank You So Much for Your time and and your comments, i will make those changes...thank you

**Pull request merged**